test/setup/main.tf

```diff
@@ -14,20 +14,25 @@
  * limitations under the License.
  */
 
-resource "random_id" "random_project_id_suffix" {
+locals {
+  random_id          = replace(lower(random_id.source.id), "/[_-]/", "t")
+  project_randomness = "${replace(substr(local.random_id, 0, 1), "/[0123456789]/", "z")}${substr(local.random_id, 1, 3)}"
+}
+
+resource "random_id" "source" {
   byte_length = 2
 }
 
 resource "google_folder" "test_folder" {
-  display_name = "test_foundation_folder_${random_id.random_project_id_suffix.hex}"
+  display_name = "test_foundation_folder_${random_id.source.hex}"
```

Comment:
I would avoid using hex if possible as that limits you to 0-9a-f rather than 0-9a-z.  Consider using `random_string` as exampled above.